src/workerd/api/web-socket.c++

```diff
@@ -58,7 +58,10 @@ WebSocket::WebSocket(jsg::Lock& js,
               package.closedOutgoingConnection)),
       outgoingMessages(IoContext::current().addObject(kj::heap<OutgoingMessagesMap>())),
       locality(LOCAL),
-      maybeCriticalSection(IoContext::current().getCriticalSection()) {}
+      maybeCriticalSection([&]() {
```

Comment:
I don't think you need to change anything in this file. `IoContext::getCriticalSection()` _already_ returns `Maybe<Own<CriticalSection>>`. So you jsut have to change the type of `maybeCriticalSection` and the existing code will already work as-is.

Reply:
I just realized as I reopened my laptop, just wanted to get something up as I was leaving a cafe, fixing now brb